### PR TITLE
CCTRI-245 Add params to POST method for adding "wait_for" functionality

### PR DESCRIPTION
**Pull request by cawabanga2811**

### CODE REVIEW

File: threatresponse/api/incident.py [hunk 1]
```diff
@@ -11,12 +11,14 @@ def __init__(self, request):
         super(IncidentAPI, self).__init__(request, '/ctia/incident')
 
     @route('status')
-    def _perform(self, id_, payload):
-        return self._post('%s/%s/status' % (self._url, id_), json=payload)
+    def _perform(self, id_, payload, **params):
```
Comment (gdavoian) [inline threatresponse/api/incident.py]: Looks that this endpoint doesn't support any additional query parameters. So let's keep it simple and use the old implementation.

File: threatresponse/api/incident.py [hunk 2]
```diff
@@ -11,12 +11,14 @@ def __init__(self, request):
...
+        return self._post('%s/%s/status' % (self._url, id_),
+                          json=payload, params=params)
 
     @route('link')
-    def _perform(self, id_, payload):
-        return self._post('%s/%s/link' % (self._url, id_), json=payload)
+    def _perform(self, id_, payload, **params):
```
Comment (gdavoian) [inline threatresponse/api/incident.py]: The same here.

File: threatresponse/api/casebook.py [hunk 3]
```diff
@@ -11,13 +11,16 @@ def __init__(self, request):
         super(CasebookAPI, self).__init__(request, '/ctia/casebook')
 
     @route('observables')
-    def _perform(self, id_, payload):
-        return self._post('%s/%s/observables' % (self._url, id_), json=payload)
+    def _perform(self, id_, payload, **params):
```
Comment (gdavoian) [inline threatresponse/api/casebook.py]: These 3 endpoints also don't have any query parameters. Also, they don't create a new resource, they modify an existing one, so there's no sense in `wait_for`.

**Comment by gdavoian** [comment]
Please make sure to add `CCTRI-245` to the PR name, since the branch will be deleted someday after merging the PR.

File: threatresponse/api/casebook.py [hunk 4]
```diff
@@ -12,12 +12,15 @@ def __init__(self, request):
...
-        return self._post('%s/%s/observables' % (self._url, id_), json=payload)
+        return self._post('%s/%s/observables' % (self._url, id_),
+                          json=payload)
 
     @route('texts')
-    def _perform(self, id_, payload):
-        return self._post('%s/%s/texts' % (self._url, id_), json=payload)
+    def _perform(self, id_, payload, **params):
+        return self._post('%s/%s/texts' % (self._url, id_),
+                          json=payload, params=params)
```
Comment (gdavoian) [inline threatresponse/api/casebook.py]: `params` aren't needed here.

File: threatresponse/api/casebook.py [hunk 5]
```diff
@@ -12,12 +12,15 @@ def __init__(self, request):
...
 
     @route('bundle')
-    def _perform(self, id_, payload):
-        return self._post('%s/%s/bundle' % (self._url, id_), json=payload)
+    def _perform(self, id_, payload, **params):
+        return self._post('%s/%s/bundle' % (self._url, id_),
+                          json=payload, params=params)
```
Comment (gdavoian) [inline threatresponse/api/casebook.py]: The same here.

Comment (cawabanga2811) [reply threatresponse/api/casebook.py]: Every comment is done.

File: tests/unit/api/assertions.py [hunk 6]
```diff
@@ -33,19 +33,19 @@ def assert_succeeds_with_get(invoke, url, id_=None, **query):
     response.json.assert_called_once_with()
 
 
-def assert_succeeds_with_post(invoke, url, payload=None):
+def assert_succeeds_with_post(invoke, url, params, payload=None):
```
Comment (konstantin-ogulchansky) [inline tests/unit/api/assertions.py]: It looks like we could remove these `assert_succeeds_with_x` methods and replace them with simple usage of `invoke`.

Comment (cawabanga2811) [reply tests/unit/api/assertions.py]: Will do it once and for all.

**Pull request merged**